bson/bsoncodec/default_value_encoders_test.go

```diff
@@ -1379,6 +1380,56 @@ func TestDefaultValueEncoders(t *testing.T) {
 				buildDocument([]byte{}),
 				nil,
 			},
+			{
+				"inline overwrite",
+				struct {
+					Foo struct {
+						A int64 `bson:",minsize"`
+						B string
+					} `bson:",inline"`
+					A int64
+				}{
+					Foo: struct {
+						A int64 `bson:",minsize"`
+						B string
+					}{
+						A: 0,
+						B: "foo",
+					},
+					A: 54321,
+				},
+				buildDocument(func(doc []byte) []byte {
+					doc = bsoncore.AppendStringElement(doc, "b", "foo")
+					doc = bsoncore.AppendInt64Element(doc, "a", 54321)
+					return doc
+				}(nil)),
+				nil,
+			},
+			{
+				"inline overwrite respects ordering",
+				struct {
+					A   int64
+					Foo struct {
+						A int64 `bson:",minsize"`
+						B string
+					} `bson:",inline"`
+				}{
+					A: 54321,
+					Foo: struct {
+						A int64 `bson:",minsize"`
+						B string
+					}{
+						A: 0,
+						B: "foo",
+					},
+				},
+				buildDocument(func(doc []byte) []byte {
+					doc = bsoncore.AppendInt64Element(doc, "a", 54321)
+					doc = bsoncore.AppendStringElement(doc, "b", "foo")
+					return doc
+				}(nil)),
+				nil,
+			},
```

Comment:
Same comment about adding a test to validate that different types for `A` are still ok.